manifests/init.pp

```diff
@@ -107,44 +119,45 @@
   $proxy_password = $pulp::params::proxy_password,
 
   $num_workers = $pulp::params::num_workers,
+  $broker_manage             = $pulp::params::broker_manage,
+  $db_manage                 = $pulp::params::db_manage,
 
   ) inherits pulp::params {
+  validate_bool($db_manage)
+  validate_bool($broker_manage)
 
   include ::apache
+  include ::mongodb::client
 
-  if (versioncmp($::mongodb_version, '2.6.5') >= 0) {
-    $mongodb_pidfilepath = '/var/run/mongodb/mongod.pid'
-  } else {
-    $mongodb_pidfilepath = '/var/run/mongodb/mongodb.pid'
+  if $db_manage {
+    include ::mongodb::server
+
+    Service['mongodb'] -> Service['pulp_celerybeat']
+    Service['mongodb'] -> Service['pulp_workers']
+    Service['mongodb'] -> Service['pulp_resource_manager']
+    Service['mongodb'] -> Exec['migrate_pulp_db']
   }
 
-  class { '::mongodb::globals':
-    version => $::mongodb_version, # taken from the custom facts
+  if $broker_manage {
+    if $messaging_transport == 'qpid' {
+      include ::qpid
+      $broker_service = 'qpidd'
+    } elsif $messaging_transport == 'rabbitmq' {
+      include ::rabbitmq
+      $broker_service = 'rabbitmq-server'
+    }
+    Service[$broker_service] -> Service['pulp_celerybeat']
+    Service[$broker_service] -> Service['pulp_workers']
+    Service[$broker_service] -> Service['pulp_resource_manager']
+  } else {
+    if $messaging_transport == 'qpid' {
+      package { 'qpid-tools': ensure => installed } -> Class['pulp::service']
+    }
   }
-  class { '::apache::mod::wsgi':} ~>
-  class { '::mongodb':
-    logpath     => '/var/lib/mongodb/mongodb.log',
-    dbpath      => '/var/lib/mongodb',
-    pidfilepath => $mongodb_pidfilepath,
-  } ~>
```

Comment:
Unfortunately, we still need this logic due to the assumptions the mongodb module is making about what is available in upstream OS repositories.